lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs

```diff
@@ -157,22 +157,58 @@ public void DeleteDirectory(Uri directoryUri)
         }
 
         /// <summary>
-        /// Gets the children of the blob "directory."
+        /// Gets the children of the container (if the uri has segments)
+        /// or containers in the storage account (if the uri has no segments)
         /// </summary>
         public IEnumerable<Uri> GetChildren(Uri directoryUri)
         {
+            string[] segments = directoryUri.Segments;
             BlobContinuationToken blobContinuationToken = null;
-            var path = directoryUri.AbsolutePath.Trim('/');
+
+            // If at the root, return all containers
+            if (segments.Count() <= 1)
+            {
+                do
+                {
+                    ContainerResultSegment containerListing = _client.ListContainersSegmented(blobContinuationToken);
+
+                    if (containerListing != null && containerListing.Results != null)
+                    {
+                        foreach (CloudBlobContainer containerItem in containerListing.Results)
+                        {
+                            yield return containerItem.Uri;
+                        }
+                    }
+
+                    blobContinuationToken = containerListing.ContinuationToken;
+                }
+                while (blobContinuationToken != null);
+                yield break;
+            }
+
+            // If not at the root folder, return all blobs within the container
+            string containerName = segments[1];
+            string relativeAddress = directoryUri.PathAndQuery.Substring(containerName.Length + 1);
 
             do
             {
-                var listing = _client.ListBlobsSegmented(path, false,
-                    BlobListingDetails.None, null,
-                    blobContinuationToken, new BlobRequestOptions(), new OperationContext());
+                BlobResultSegment listing = _client.ListBlobsSegmented(
+                    containerName,
+                    relativeAddress,
+                    useFlatListing: false,
+                    BlobListingDetails.None,
+                    maxResults: null,
+                    blobContinuationToken,
+                    new BlobRequestOptions(),
+                    new OperationContext());
 
-                if (listing.Results != null)
+                if (listing == null || listing.Results.Count() == 0)
+                {
+                    throw new ArgumentException("Call to ListBlobsSegmented returned no results. Uri is invalid or does not have children.", nameof(directoryUri));
```

Comment:
this line is quite long. our coding standard requires that lines should be no longer than 120 characters; we usually keep it below 100 for readability.